File: Python/Torres_Hanoi.py
movimientos = 0

def torres_hanoi(n_discos: int, torre_inicio: str, torre_final: str, torre_pivote: str):
    global movimientos
    if(n_discos == 1):
        movimientos +=1
        print(f"{movimientos}: Mover disco 1 de {torre_inicio} a {torre_final}")
        return
    torres_hanoi(n_discos-1, torre_inicio, torre_pivote, torre_final)
    movimientos +=1
    print(f"{movimientos}: Mover disco {n_discos} de {torre_inicio} a {torre_final}")
    torres_hanoi(n_discos-1, torre_pivote, torre_final, torre_inicio)

torres_hanoi(7,'A','C','B')

File: C/Torres_Hanoi.c
#include <stdio.h>

void torresHanoi(int n_discos, char torreInicio, char torreFinal, char torrePivote);

int main() {
    int n_discos = 7;
    char torreInicio = 'A';
    char torreFinal = 'C';
    char torrePivote = 'B';
    torresHanoi(n_discos, torreInicio, torreFinal, torrePivote);

    return 0;
}

void torresHanoi(int n_discos, char torreInicio, char torreFinal, char torrePivote) {
    if(n_discos == 1) {
        printf("Mover disco 1 de %c a %c\n", torreInicio, torreFinal);
        return;
    }
    torresHanoi(n_discos-1, torreInicio, torrePivote, torreFinal);
    printf("Mover disco %d de %c a %c\n", n_discos, torreInicio, torreFinal);
    torresHanoi(n_discos-1, torrePivote, torreFinal, torreInicio);
}